### PR TITLE
Preserve the final job result's post condition in streaming scenario.

### PR DESCRIPTION
**Pull request by Jorghi12**
After conversing with Justin, we have opted to combine streaming and non-streaming functionalities into a unified worker image. Furthermore, the resulting output can be consolidated into a single job output, tentatively labeled as `submit_stream_on_end`.

### CODE REVIEW

File: runpod/serverless/worker.py [hunk 1]
```diff
@@ -70,9 +70,14 @@ async def process_job(job):
                     job_result = run_job_generator(config["handler"], job)
 
                     log.debug("Handler is a generator, streaming results.")
+                    job_outputs = []
                     async for job_stream in job_result:
+                        if config["submit_stream_on_end"]:
```
Comment (justinmerrell) [inline runpod/serverless/worker.py]: a slight name change, still open to suggestions but something that has less ambiguity with that the result will be "return_aggregate_stream"

File: runpod/serverless/worker.py [hunk 2]
```diff
@@ -70,9 +70,14 @@ async def process_job(job):
                     job_result = run_job_generator(config["handler"], job)
 
                     log.debug("Handler is a generator, streaming results.")
+                    job_outputs = []
```
Comment (justinmerrell) [inline runpod/serverless/worker.py]: job_results = {'output': []}

File: runpod/serverless/worker.py [hunk 3]
```diff
@@ -70,9 +70,14 @@ async def process_job(job):
...
                     async for job_stream in job_result:
+                        if config["submit_stream_on_end"]:
+                            job_outputs.append(job_stream)
```
Comment (justinmerrell) [inline runpod/serverless/worker.py]: job_results['output'].append(job_stream)

File: runpod/serverless/worker.py [hunk 4]
```diff
@@ -70,9 +70,14 @@ async def process_job(job):
...
                         await stream_result(session, job_stream, job)
                     job_result = {}
```
Comment (justinmerrell) [inline runpod/serverless/worker.py]: Can these lines now be eliminated?

File: runpod/serverless/worker.py (hunk 1 above)
Comment (Jorghi12) [reply runpod/serverless/worker.py]: return_aggregate_stream is a better name. sg.

**Review by justinmerrell — APPROVED** [review]

**Pull request merged**